packages/lib/src/vis/line/hooks.ts

```diff
@@ -1,60 +1,6 @@
-import type { NumArray } from '@h5web/shared';
-import { useMemo } from 'react';
-import { Vector3 } from 'three';
+import { createMemo } from '@h5web/shared';
 
-import { useVisCanvasContext } from '../shared/VisCanvasProvider';
+import { getValueToErrorPositions, getValueToPosition } from './utils';
 
-const CAMERA_FAR = 1000; // R3F's default
-
-export function useCanvasPoints(
-  abscissas: NumArray,
-  ordinates: NumArray,
-  errors?: NumArray,
-  ignoreValue?: (val: number) => boolean
-) {
-  const { abscissaScale, ordinateScale } = useVisCanvasContext();
-
-  return useMemo(() => {
-    const dataPoints: Vector3[] = [];
-    const errorBarSegments: Vector3[] = [];
-    const errorCapPoints: Vector3[] = [];
-
-    ordinates.forEach((val, index) => {
-      const x = abscissaScale(abscissas[index]);
-      const y = ordinateScale(val);
-
-      const hasFiniteCoords =
-        !ignoreValue?.(val) && Number.isFinite(x) && Number.isFinite(y);
-      const dataVector = hasFiniteCoords
-        ? new Vector3(x, y, 0)
-        : /* Render points with NaN/Infinity coordinates (i.e. values <= 0 in log)
-           * at origin to avoid Three warning, and outside of camera's field of view
-           * to hide them and any segments connecting them. */
-          new Vector3(0, 0, CAMERA_FAR);
-
-      dataPoints.push(dataVector);
-
-      const error = errors?.[index];
-      if (!error || !hasFiniteCoords) {
```

Comment:
This condition is why error bars where not being rendered before for error values equal to `0`.